index.js

```diff
@@ -66,21 +77,23 @@ module.exports = (function () {
 	return lazyLoad;
 }());
 
+/** @type {Tape['createHarness']} */
 function createHarness(conf_) {
 	var results = new Results({ todoIsOK: !!(process.env.TODO_IS_OK === '1') });
 	if (!conf_ || conf_.autoclose !== false) {
 		results.once('done', function () { results.close(); });
 	}
 
+	/** @type {(name: string, conf: import('.').TestOptions, cb: Test.Callback) => Test} */
 	function test(name, conf, cb) {
 		var t = new Test(name, conf, cb);
 		test._tests.push(t);
 
 		(function inspectCode(st) {
-			st.on('test', function sub(st_) {
+			st.on('test', /** @type {(st: Test) => void} */ function sub(st_) {
 				inspectCode(st_);
 			});
-			st.on('result', function (r) {
+			st.on('result', /** @type {(r: import('./lib/results').Result) => void} */ function (r) {
```

Comment:
> `import('./lib/results').Result`

Is there a reason not to put `Result` in `index.d.ts` and import it from there?

Reply:
it felt like the proper place to put it was with the Results class